src/scripts/run-tests.sh

```diff
@@ -11,26 +20,36 @@ unzip -qod "${tmpdir}/scriptgen" "${tmpdir}/scriptgen.zip"
 # form OS appropriate paths for MATLAB
 os=$(uname)
 gendir=$tmpdir
+workdir=$(pwd)
+binext=""
 if [[ $os = CYGWIN* || $os = MINGW* || $os = MSYS* ]]; then
     gendir=$(cygpath -w "$gendir")
+    workdir=$(cygpath -w "$workdir")
+    binext=".exe"
 fi
 
 # generate and run MATLAB test script
-"${tmpdir}/bin/run_matlab_command.sh" "\
-    addpath('${gendir}/scriptgen');\
-    testScript = genscript('Test',\
-    'JUnitTestResults','${PARAM_TEST_RESULTS_JUNIT}',\
-    'CoberturaCodeCoverage','${PARAM_CODE_COVERAGE_COBERTURA}',\
-    'HTMLCodeCoverage','${PARAM_CODE_COVERAGE_HTML}',\
-    'SourceFolder','${PARAM_SOURCE_FOLDER}',\
-    'SelectByFolder','${PARAM_SELECT_BY_FOLDER}',\
-    'SelectByTag','$PARAM_SELECT_BY_TAG',\
-    'CoberturaModelCoverage','${PARAM_MODEL_COVERAGE_COBERTURA}',\
-    'HTMLModelCoverage','${PARAM_MODEL_COVERAGE_HTML}',\
-    'SimulinkTestResults','${PARAM_TEST_RESULTS_SIMULINK_TEST}',\
-    'HTMLTestReport','${PARAM_TEST_RESULTS_HTML}',\
-    'PDFTestReport','${PARAM_TEST_RESULTS_PDF}');\
-    disp('Running MATLAB script with contents:');\
-    disp(testScript.Contents);\
-    fprintf('__________\n\n');\
-    run(testScript);"
+script=command_${RANDOM}
+scriptpath=${tmpdir}/${script}.m
+echo "cd('${workdir//\'/\'\'}');" > "$scriptpath"
+cat << EOF >> "$scriptpath"
+addpath('${gendir}/scriptgen');\
+testScript = genscript('Test',\
+'JUnitTestResults','${PARAM_TEST_RESULTS_JUNIT}',\
+'CoberturaCodeCoverage','${PARAM_CODE_COVERAGE_COBERTURA}',\
+'HTMLCodeCoverage','${PARAM_CODE_COVERAGE_HTML}',\
+'SourceFolder','${PARAM_SOURCE_FOLDER}',\
+'SelectByFolder','${PARAM_SELECT_BY_FOLDER}',\
+'SelectByTag','$PARAM_SELECT_BY_TAG',\
+'CoberturaModelCoverage','${PARAM_MODEL_COVERAGE_COBERTURA}',\
+'HTMLModelCoverage','${PARAM_MODEL_COVERAGE_HTML}',\
+'SimulinkTestResults','${PARAM_TEST_RESULTS_SIMULINK_TEST}',\
+'HTMLTestReport','${PARAM_TEST_RESULTS_HTML}',\
+'PDFTestReport','${PARAM_TEST_RESULTS_PDF}');\
+disp('Running MATLAB script with contents:');\
+disp(testScript.Contents);\
+fprintf('__________\n\n');\
+run(testScript);
+EOF
```

Comment:
Untested, but think this should work?

```suggestion
cat << EOF > "$scriptpath"
cd('${workdir//\'/\'\'}');
addpath('${gendir}/scriptgen');
testScript = genscript('Test', ...
    'JUnitTestResults','${PARAM_TEST_RESULTS_JUNIT}', ...
    'CoberturaCodeCoverage','${PARAM_CODE_COVERAGE_COBERTURA}', ...
    'HTMLCodeCoverage','${PARAM_CODE_COVERAGE_HTML}', ...
    'SourceFolder','${PARAM_SOURCE_FOLDER}', ...
    'SelectByFolder','${PARAM_SELECT_BY_FOLDER}', ...
    'SelectByTag','$PARAM_SELECT_BY_TAG', ...
    'CoberturaModelCoverage','${PARAM_MODEL_COVERAGE_COBERTURA}', ...
    'HTMLModelCoverage','${PARAM_MODEL_COVERAGE_HTML}', ...
    'SimulinkTestResults','${PARAM_TEST_RESULTS_SIMULINK_TEST}', ...
    'HTMLTestReport','${PARAM_TEST_RESULTS_HTML}', ...
    'PDFTestReport','${PARAM_TEST_RESULTS_PDF}');
disp('Running MATLAB script with contents:');
disp(testScript.Contents);
fprintf('__________\n\n');
run(testScript);
EOF
```

Reply:
Oh no, I had just committed a change removing heredoc when I saw this. I don't have a super strong difference either way. I think without heredoc is slightly more simple, but with heredoc is more consistent with our other integrations.

Reply:
I could go either way as well

Reply:
I'm thinking maybe don't use heredoc until if/ when we need it